jmx-metrics/src/integrationTest/java/io/opentelemetry/contrib/jmxmetrics/AbstractIntegrationTest.java

```diff
@@ -0,0 +1,233 @@
+
+  protected void assertGauge(Metric metric, String name, String description) {
+    assertGauge(metric, name, description, "µs");
+  }
+
+  protected void assertGauge(Metric metric, String name, String description, String unit) {
+    assertThat(metric.getName()).isEqualTo(name);
+    assertThat(metric.getDescription()).isEqualTo(description);
+    assertThat(metric.getUnit()).isEqualTo(unit);
+    assertThat(metric.hasGauge()).isTrue();
+    assertThat(metric.getGauge().getDataPointsList())
+        .satisfiesExactly(point -> assertThat(point.getAttributesList()).isEmpty());
+  }
+
+  protected void assertSum(Metric metric, String name, String description) {
+    assertSum(metric, name, description, "1");
+  }
+
+  protected void assertSum(Metric metric, String name, String description, String unit) {
+    assertThat(metric.getName()).isEqualTo(name);
+    assertThat(metric.getDescription()).isEqualTo(description);
+    assertThat(metric.getUnit()).isEqualTo(unit);
+    assertThat(metric.hasSum()).isTrue();
+    assertThat(metric.getSum().getDataPointsList())
+        .satisfiesExactly(point -> assertThat(point.getAttributesList()).isEmpty());
+  }
+
+  protected void assertTypedGauge(
+      Metric metric, String name, String description, String unit, List<String> types) {
+    assertThat(metric.getName()).isEqualTo(name);
+    assertThat(metric.getDescription()).isEqualTo(description);
+    assertThat(metric.getUnit()).isEqualTo(unit);
+    assertThat(metric.hasGauge()).isTrue();
+    assertTypedPoints(metric.getGauge().getDataPointsList(), types);
+  }
+
+  protected void assertTypedSum(
+      Metric metric, String name, String description, String unit, List<String> types) {
+    assertThat(metric.getName()).isEqualTo(name);
+    assertThat(metric.getDescription()).isEqualTo(description);
+    assertThat(metric.getUnit()).isEqualTo(unit);
+    assertThat(metric.hasSum()).isTrue();
+    assertTypedPoints(metric.getSum().getDataPointsList(), types);
+  }
+
+  @SuppressWarnings("unchecked")
+  private static void assertTypedPoints(List<NumberDataPoint> points, List<String> types) {
+    assertThat(points)
+        .satisfiesExactlyInAnyOrder(
+            types.stream()
+                .map(
+                    type ->
+                        (Consumer<NumberDataPoint>)
+                            point ->
+                                assertThat(point.getAttributesList())
+                                    .singleElement()
+                                    .satisfies(
+                                        attribute -> {
+                                          assertThat(attribute.getKey()).isEqualTo("name");
+                                          assertThat(attribute.getValue().getStringValue())
+                                              .isEqualTo(type);
+                                        }))
+                .toArray(Consumer[]::new));
+  }
+
+  private static class OtlpGrpcServer extends ServerExtension {
```

Comment:
I realized these sort of integration tests are tedious outside the instrumentation repo. Providing an artifact that has an OTLP receiver that maps back to SDK types, to take advantage of SDK assertions like the instrumentation repo is able to, will probably be nice at some point.

Reply:
https://github.com/open-telemetry/opentelemetry-java-contrib/issues/24 could be satisfied by that aim, but these changes may be adequate as well.